site/scripts/prepare-blocks.ts

```diff
@@ -117,26 +119,18 @@ const ensureRepositorySnapshot = async ({
   commit: string;
   workshopDirPath: string;
 }): Promise<string> => {
-  if (
-    !repositoryHref.match(/^https:\/\/github.com\/[\w-]+\/[\w-]+(\/|.git)?$/i)
-  ) {
-    if (repositoryHref.startsWith("https?://github.com")) {
-      throw new Error(
-        `Cannot handle repository URL ${repositoryHref}. Only GitHub links are currently supported. We will add more services based on the demand.`,
-      );
-    } else {
-      throw new Error(
-        `Cannot handle repository URL ${repositoryHref}. It needs to match https://github.com/org/repo`,
-      );
-    }
+  const tarballUrl = hostedGitInfo
```

Comment:
We generate the `tarball` URL in a one-shot fashion and use it throughout the code below.